src/nvim/decoration.c

```diff
@@ -346,7 +346,14 @@ char *next_virt_text_chunk(VirtText vt, size_t *pos, int *attr)
   for (; text == NULL && *pos < kv_size(vt); (*pos)++) {
     text = kv_A(vt, *pos).text;
     int hl_id = kv_A(vt, *pos).hl_id;
-    *attr = hl_combine_attr(*attr, hl_id > 0 ? syn_id2attr(hl_id) : 0);
+    if (attr != NULL) {
+      *attr = hl_combine_attr(*attr, hl_id > 0 ? syn_id2attr(hl_id) : 0);
+      if (*attr == 0) {
+        // If no highlight is detected, set attr to -1 to indicate missing highlight
+        // for subsequent highlight detection.
+        *attr = -1;
+      }
```

Comment:
I don't think this is correct. How does this prevent overriding when `hl_id` is 0?